app/assets/javascripts/shared/editor_toolbar.js

```diff
@@ -15,18 +15,28 @@ class EditorToolbar {
     if (!$target.is("textarea")) { console.log("Can't initialize a rich toolbar on anything but a textarea"); return; }
 
     this.$target = $target;
-    this.opts = { 'include': $target.data('rich-toolbar').split(' ') };
+    this.opts = {
+      'include': $target.data('rich-toolbar').split(' '),
+      'uploader': $target.data('rich-toolbar-uploader')
+    };
+
+    if (this.opts.include.includes('image') && this.opts.uploader === undefined) { console.log("You initialized a RichToolbar with the image uploader option but have not provided an existing uploader to utilize"); return; }
+
     this.affixes = this.affixesLibrary();
     this.init();
   }
 
   init() {
     this.$target.wrap('<div class="editor-field" data-behavior="editor-field"><div class="textarea-container"></div></div>');
-    this.$editorField = this.$target.parents('[data-behavior=editor-field]');
-    this.$editorField.prepend('<div class="editor-toolbar" data-behavior="editor-toolbar"></div>');
-    this.$editorToolbar = this.$editorField.find('[data-behavior=editor-toolbar]');
+    this.$editorField = this.$target.parents('[data-behavior~=editor-field]');
+    this.$editorField.prepend('<div class="editor-toolbar" data-behavior="editor-toolbar" data-edit="#'+ this.$target.data('id') + '"></div>');
+    this.$editorToolbar = this.$editorField.find('[data-behavior~=editor-toolbar]');
 
     this.$editorToolbar.append(this.textareaElements(this.opts.include));
+    this.$fileField = $('<input type="file" name="editor-toolbar-' + Math.random().toString(36) + '[]" multiple style="display: none">');
```

Comment:
What does the `name="editor-toolbar-' + Math.random().toString(36) + '[]` name do? Don't see it used anywhere

Reply:
It's designed to assign random names to the input field, ensuring no conflict as an input would exist for every field on the page.

Reply:
Added `accept="image/*"` on the toolbar button. Technically the attachment has always supported any attachment kind so I've not enforced it any further. You could drag and drop another file type, but the `Image` button on the toolbar only supports images.